ibis/backends/base/__init__.py

```diff
@@ -369,6 +369,100 @@ def read_csv(
             f"{self.name} does not support direct registration of CSV data."
         )
 
+    @util.experimental
+    def to_parquet(
+        self,
+        expr: ir.Table,
+        path: str | Path,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Write the results of executing the given expression to a parquet file.
+
+        This method is eager and will execute the associated expression
+        immediately.
+
+        Parameters
+        ----------
+        expr
+            The ibis expression to execute and persist to parquet.
+        path
+            The data source. A string or Path to the parquet file.
+        params
+            Mapping of scalar parameter expressions to value.
+        limit
+            An integer to effect a specific row limit. A value of `None` means
+            "no limit". The default is in `ibis/config.py`.
+        **kwargs
+            Additional keyword arguments passed to pyarrow.parquet.ParquetWriter
+
+        https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html
+        """
+        self._import_pyarrow()
+        import pyarrow.parquet as pq
+
+        batch_reader = expr.to_pyarrow_batches(params=params, limit=limit)
+
+        writer = pq.ParquetWriter(path, batch_reader.schema)
```

Comment:
Can we use the `*Writer`s as context managers to avoid the `try:`/`finally:`?

Reply:
We can use them as context managers, but the record batch reader will still raise a `StopIteration` when it's exhausted.